zerver/tests/test_message_fetch.py

```diff
@@ -3483,18 +3483,25 @@ def test_bad_int_params(self) -> None:
         other_params = {"narrow": {}, "anchor": 0}
         int_params = ["num_before", "num_after"]
 
-        bad_types = (False, "", "-1", -1)
+        bad_types: List[Dict[str, Any]] = [
+            {"type": False, "error_msg": "Bad value for"},
+            {"type": "", "error_msg": "Bad value for"},
+            {"type": "-1", "error_msg": "Bad value for"},
+            {"type": -1, "error_msg": "Bad value for"},
+        ]
         for idx, param in enumerate(int_params):
-            for type in bad_types:
+            for type_dict in bad_types:
```

Comment:
The `type_dict/bad_types` names feel fairly confusing at this point. 

Maybe call these `invalid_parameters`, and `invalid_parameter["value"]/invalid_parameter["expected_error"]`? That seems a lot more readable.

Reply:
`Dict[str, Any]` is not as restrictive as a dataclass so it is less preferrable. We could have something like:

```python
@dataclass
class InvalidParam:
    value: object
    expected_error: str
```

This can also be reused for later occurrences of `bad_types` in other tests.